docs/deployment/k8s.md

```diff
@@ -0,0 +1,519 @@
+dynamically. Therefore we don't want to try to connect to pods directly, but
+rather create a Kubernetes Service. Services keep track of pods and direct
+traffic to the right place.
+
+The final step for our database is to create the service descriptor:
+
+```yaml
+# kubernetes/postgres-service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres
+  namespace: backstage
+spec:
+  selector:
+    app: postgres
+  ports:
+    - port: 5432
+```
+
+Apply the service to the Kubernetes cluster:
+
+```shell
+$ kubectl apply -f kubernetes/postgres-service.yaml
+service/postgres created
+
+$ kubectl get services --namespace=backstage
+NAME       TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)    AGE
+postgres   ClusterIP   10.96.5.103   <none>        5432/TCP   29s
+```
+
+## Creating the Backstage instance
+
+Now that we have PostgreSQL up and ready to store data, we can create the
+Backstage instance. This follows similar steps as the PostgreSQL deployment.
+
+### Creating a Backstage secret
+
+For any Backstage configuration secrets, such as authorization tokens, we can
+create a similar Kubernetes Secret as we did
+[for PostgreSQL](#creating-a-postgresql-secret), remembering to base64 encode
+the values:
+
+```yaml
+# kubernetes/backstage-secrets.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: backstage-secrets
+  namespace: backstage
+type: Opaque
+data:
+  GITHUB_TOKEN: VG9rZW5Ub2tlblRva2VuVG9rZW5NYWxrb3ZpY2hUb2tlbg==
+```
+
+Apply the secret to the Kubernetes cluster:
+
+```shell
+$ kubectl apply -f kubernetes/backstage-secrets.yaml
+secret/backstage-secrets created
+```
+
+### Creating a Backstage deployment
+
+To create the Backstage deployment, first create a [Docker image](docker.md).
+We'll use this image to create a Kubernetes deployment. For this example, we'll
+use the standard host build with the frontend bundled and served from the
+backend.
+
+First, create a Kubernetes Deployment descriptor:
+
+```yaml
+# kubernetes/backstage.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: backstage
+  namespace: backstage
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: backstage
+  template:
+    metadata:
+      labels:
+        app: backstage
+    spec:
+      containers:
+        - name: backstage
+          image: backstage:1.0.0
+          imagePullPolicy: IfNotPresent
+          ports:
+            - name: http
+              containerPort: 7000
+          envFrom:
+            - secretRef:
+                name: postgres-secrets
+            - secretRef:
+                name: backstage-secrets
+```
+
+For production deployments, the `image` reference will usually be a full URL to
+a repository on a container registry (for example, ECR on AWS).
+
+For testing locally with `minikube`, you can point the local Docker daemon to
+the `minikube` internal Docker registry and then rebuild the image to install
+it:
+
+```shell
+$ eval $(minikube docker-env)
+$ yarn build-image --tag backstage:1.0.0
```

Comment:
Is this meant to be run from the top level of the directory?  Should this be `yarn docker-build --tag backstage:1.0.0` ?

Reply:
This is for a Backstage app made with `npx @backstage/create-app`, which has a `build-image` target. The `docker-build` target is valid in the main repo, but doesn't exist in `create-app`. A bit confusing, could use to sync these up.